tests/smoke/ckeditor.js

```diff
@@ -1,28 +1,59 @@
-/* bender-tags: editor */
-/* bender-ckeditor-plugins: wysiwygarea,structuredheadings */
+/* bender-tags: editor,unit */
+/* bender-ckeditor-plugins: wysiwygarea,structuredheadings,toolbar,basicstyles */
 
 // Clean up all instances been created on the page.
 (function () {
-  var removeAllInstances = () => {
-    var allInstances = CKEDITOR.instances;
-    for (var i in allInstances) {
-      CKEDITOR.remove(allInstances[i]);
-    }
+
+  bender.editor = {
+    allowedForTests: "h1; p"
   };
 
   bender.test({
-    setUp: () => {
-      removeAllInstances();
+    setUp: function () {
+      //Anything to be run before each test if needed
     },
 
-    "CKEditor Loads": () => {
+    "CKEditor Loads": function () {
       CKEDITOR.replace("editor1");
       assert.isObject(CKEDITOR.instances.editor1, "editor instance not found");
     },
-    
-    "Structured Headings Plugin Available": () => {
-    	assert.isObject(CKEDITOR.plugins.get('structuredheadings'), "plugin not available");
+
+    "Structured Headings Plugin Available": function () {
+      assert.isObject(CKEDITOR.plugins.get("structuredheadings"), "plugin not available");
+    },
+
+    "Command Active but Off Inside Header Tag": function () {
+      this.editorBot.setHtmlWithSelection("<h1>^Heading</h1>");
+      var testCommand = this.editor.getCommand("autoNumberHeading");
+
+      assert.areSame(
+        CKEDITOR.TRISTATE_OFF, testCommand.state,
+        "autoNumberHeading OFF in H1"
+      );
+
+    },
+
+    "Command Active and On Inside AutoNumbered Header": function () {
+      this.editorBot.setHtmlWithSelection("<h1 class='autonumber'>^Heading</h1>");
+      var testCommand = this.editor.getCommand("autoNumberHeading");
+
+      assert.areSame(
+        CKEDITOR.TRISTATE_ON, testCommand.state,
+        "autoNumberHeading ON in H1"
+      );
+
+    },
+
+    "Command Disabled Outside Header": function () {
+      this.editorBot.setHtmlWithSelection("<p>^Paragraph</p>");
+      var testCommand = this.editor.getCommand("autoNumberHeading");
+
+      assert.areSame(
+        CKEDITOR.TRISTATE_DISABLED, testCommand.state,
+        "autoNumberHeading DISABLEd in P"
+      );
+
```

Comment:
These tests look good for testing command state (actually, we might want to break these out into a separate file, tests/commandstate.js or something.

I'm a bit surprised we don't have tests for actually `execCommand`  though.  Can we perhaps add tests for when we start in off/on state and then toggle using the command?  Similarly there's also a bot.editorBot.getHtmlWithSelection that does the same thing in reverse - we can assert on the expected html this way.